src/ensembl/production/djcore/models.py

```diff
@@ -43,27 +43,24 @@ class NullTextField(models.TextField):
     def __init__(self, *args, **kwargs):
         kwargs['null'] = True
         kwargs['blank'] = True
+        self.trim_cr = kwargs.pop('trim_cr', False)
         super(NullTextField, self).__init__(*args, **kwargs)
 
     def to_python(self, value):
         if isinstance(value, models.CharField):
-            return value
-        if value == None:
+            return value if not self.trim_cr else trim_carriage_return(value)
+        if value is None:
             return ""
         else:
-            return value # otherwise, return just the value
+            return value if not self.trim_cr else trim_carriage_return(value)
 
     def get_db_prep_value(self, value, connection, prepared=False):
-        print("value is [%s]" % value)
         if not value:
             # if Django tries to save an empty string, send to db None (NULL)
             return None
         else:
-            return value # otherwise, just pass the value
-        return super().get_db_prep_value(value, connection, prepared)
-
-    def get_internal_type(self):
-        return "TextField"
+            value = value if not self.trim_cr else trim_carriage_return(value)
```

Comment:
This is a bit convoluted, I think the following is more readable

```suggestion
            if self.trim_cr:
                value = trim_carriage_return(value)
```